nada_dsl/nada_types/scalar_types.py

```diff
@@ -161,6 +161,19 @@ def __ge__(self, other):
             "GreaterOrEqualThan", ">=", self, other, lambda lhs, rhs: lhs >= rhs
         )
 
+    def __radd__(self, other):
+        """This adds support for builtin `sum` operation for numeric types."""
+        if isinstance(other, int):
+            other_type = NumericType(
+                Literal(value=other, source_ref=SourceRef.back_frame()),
+                base_type=self.base_type,
+                mode=Mode.CONSTANT,
+            )
```

Comment:
Shouldn't you use `new_scalar_type(mode, base_type)(Literal(value=other, source_ref=SourceRef.back_frame()))` to create the specific instance of Integer or UnsingedInteger?

Reply:
In addition, I would say that if you use the factory I mentioned above, the following line (`other_type.value = other`) is not necessary

Reply:
I wasn't familiar with this constructor, thank you!

Reply:
Actually your proposal is not correct, it leads to wrong results. Please reconsider your comment.

Reply:
I'm sorry, the correct replacement should be `new_scalar_type(Mode.CONSTANT, base_type)(Literal(value=other, source_ref=SourceRef.back_frame())`